sdk/python/tests/integration/offline_store/test_universal_historical_retrieval.py

```diff
@@ -216,18 +251,30 @@ def test_historical_features(environment, universal_data_sources, full_feature_n
         i for i in range(len(entity_df_with_request_data))
     ]
 
-    customer_fv, driver_fv, driver_odfv, order_fv, global_fv = (
+    customer_fv, driver_fv, driver_odfv, location_fv, order_fv, global_fv = (
         feature_views["customer"],
         feature_views["driver"],
         feature_views["driver_odfv"],
+        feature_views["location"],
         feature_views["order"],
         feature_views["global"],
     )
 
     feature_service = FeatureService(
-        "convrate_plus100",
+        name="convrate_plus100",
         features=[feature_views["driver"][["conv_rate"]], feature_views["driver_odfv"]],
     )
+    feature_service_shadow_entities = FeatureService(
+        name="shadow_entities",
+        features=[
+            location_fv.with_name("origin").with_join_key_map(
```

Comment:
This (and every other usage of `FV.with_name()` is dependent on https://github.com/feast-dev/feast/pull/1872.